contracts/src/lib.rs

```diff
@@ -23,4 +36,11 @@ pub extern fn call() {
     write(&x_key, &s);
     let s_read = read(&x_key);
     assert_eq!(s, s_read);
+
+    let export_name = String::from("plus_one_ext");
+    let hash = store_function(&export_name);
+    let contract = read(&hash);
+    let args = vec![3i32.to_bytes()];
```

Comment:
Rust doesn't allow overloading a function definition, so these would all need to be in different modules or have different names. I think the best we could do is provide `Into<Vec<Vec<u8>>>` instances for various tuples so that something like `let args: Vec<Vec<u8>> = (1, "hello world").into()` would work.

Reply:
Sorry for not investigating it before writing. Apparently it is difficult but doable: https://www.reddit.com/r/rust/comments/2umcxv/wait_rust_doesnt_have_function_overloading/

Reply:
And this: https://internals.rust-lang.org/t/justification-for-rust-not-supporting-function-overloading-directly/7012/2

Reply:
Hmm, using a trait to get something that looks like overloading...interesting.
Implementing `Into`/`From` on tuples actually doesn't seem to work either, the compiler complains at me no matter how I do it. I think perhaps the actual best thing we could do is implement an `args!` macro that takes a variable number of arguments and constructs the `Vec<Vec<u8>>` from them. But I don't know how to write macros yet, so maybe I'll just leave this as a todo

Reply:
The macro would have to be called in the contract code right? Let's not do it. I think that using traits, if possible, is better choice.

Reply:
What's wrong with calling a macro in the contract code? It would work essentially the same way as the `vec!` macro does from the alloc library.